### PR TITLE
CI: add hardware reflasher

### PR DESCRIPTION
**Pull request by briskspirit**
Will reset all pandas in CI with DFU flash

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -14,6 +14,22 @@ pipeline {
         }
       }
     }
+    stage('reset hardware') {
```
Comment (adeebshihadeh) [inline Jenkinsfile:17]: need to lock all the resources and do everything at once now

File: tests/ci_resetter.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    self.enable_power(3, False)
+    time.sleep(1)
+    self.enable_power(1, True)
+    self.enable_power(2, True)
+    self.enable_power(3, True)
+    if delay > 0:
+      time.sleep(delay)
+
+if __name__ == "__main__":
+  r = Resetter()
```
Comment (adeebshihadeh) [inline tests/ci_resetter.py:58]: this part should be split out from the file with the class. there should be a file for the support lib and another to get the CI zoo in a good state. I would also move things like the test jungle and panda serials in the prep script so they're all in one place

File: tests/ci_resetter.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  print(pandas)
+  assert len(pandas) == 7
+
+  for serial in pandas:
+    p = PandaDFU(serial)
+    p.recover()
+
+  r.cycle_power(5)
+
+  pandas = [x for x in Panda.list() if x != 'none'] # Exclude pedal
```
Comment (adeebshihadeh) [inline tests/ci_resetter.py:74]: should we just exclude pedals by default in Panda.list()? this none thing is a pretty bad convention

File: Jenkinsfile (hunk 1 above)
Comment (briskspirit) [reply Jenkinsfile:17]: I am already locking those with extra, no?

File: tests/ci_resetter.py (hunk 3 above)
Comment (briskspirit) [reply tests/ci_resetter.py:74]: agree, add here or separate PR?

File: Jenkinsfile (hunk 1 above)
Comment (briskspirit) [reply Jenkinsfile:17]: `lock(extra: [[resource: "pedal"]], resource: "pandas") {`

Comment (adeebshihadeh) [reply Jenkinsfile:17]: that's only for the reset stage. the rest of the stages should all be under one lock now

File: tests/ci_resetter.py (hunk 3 above)
Comment (adeebshihadeh) [reply tests/ci_resetter.py:74]: another PR

File: tests/ci_reset_hw.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from panda import Panda, PandaDFU
+from libs.resetter import Resetter  # pylint: disable=import-error
+
+r = Resetter()
```
Comment (adeebshihadeh) [inline tests/ci_reset_hw.py:4]: always guard with `if __name__ == '__main__':`. you don't want this to run at import time

File: tests/ci_reset_hw.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+from panda import Panda, PandaDFU
+from libs.resetter import Resetter  # pylint: disable=import-error
```
Comment (adeebshihadeh) [inline tests/ci_reset_hw.py:2]: shouldn't need a pylint exception here

Comment (briskspirit) [reply tests/ci_reset_hw.py:2]: pre-commit hook and pylint tells that I need it

**Pull request merged**